public class Producer implements Runnable {

	private Counter counter;

	public Producer(Counter counter) {
		this.counter = counter;
	}

	public void run() {
		for (int i = 0; i < 1000; i++) {
			counter.increment();
		}
	}
}
